run_node.sh

```diff
@@ -2,8 +2,390 @@
+                -p "${RPC_PORT_MAP}" \
+                -p "${PORT_MAP}" \
+                -p "${VALIDATOR_PORT_MAP}" \
+                -p "${METRICS_PORT_MAP}" \
+                -u "$(id -u):$(id -g)" \
+                --mount type=bind,source="${HOST_BASE_PATH}",target=${BASE_PATH} \
+                --name "${NAME}" \
+                --restart unless-stopped \
+                -d "${ALEPH_IMAGE}"
+}
+
+run_archivist () {
+    if [ -n "${MAINNET}" ] && [ "${MAINNET}" = 'true' ]
+    then
+        source env/archivist_mainnet
+        eval "echo \"$(cat env/archivist_mainnet)\"" > env/archivist_mainnet.env
+        ENV_FILE="env/archivist_mainnet.env"
+    else
+        source env/archivist
+        eval "echo \"$(cat env/archivist)\"" > env/archivist.env
+        ENV_FILE="env/archivist.env"
+    fi
+
+
+    PROXY_PORT=${PROXY_PORT:-$PORT}
+
+    if [[ -n "${PUBLIC_DNS}" ]]
+    then
+        PUBLIC_ADDR="/dns4/${PUBLIC_DNS}/tcp/${PROXY_PORT}"
+    else
+        PUBLIC_ADDR="/ip4/${PUBLIC_IP}/tcp/${PROXY_PORT}"
+    fi
+
+    echo "Running with public P2P address: ${PUBLIC_ADDR}"
+
+    PORT_MAP="${PORT}:${PORT}"
+
+    docker run --env-file ${ENV_FILE} \
+                --env PUBLIC_ADDR="${PUBLIC_ADDR}" \
+                -p ${RPC_PORT_MAP} \
+                -p ${PORT_MAP} \
+                -p ${METRICS_PORT_MAP} \
+                -u "$(id -u):$(id -g)" \
+                --mount type=bind,source=${HOST_BASE_PATH},target=${BASE_PATH} \
+                --name ${NAME} \
+                --restart unless-stopped \
+                -d "${ALEPH_IMAGE}"
+}
+
+
+# The defaults
+NAME="aleph-node-$(xxd -l "16" -p /dev/urandom | tr -d " \n" ; echo)"
+BASE_PATH="/data"
+HOST_BASE_PATH="${HOME}/.alephzero"
+DB_SNAPSHOT_FILE="db_backup.tar.gz"
+DB_SNAPSHOT_URL="http://db.test.azero.dev.s3-website.eu-central-1.amazonaws.com/latest.html"
+MAINNET_DB_SNAPSHOT_URL="http://db.azero.dev.s3-website.eu-central-1.amazonaws.com/latest.html"
+DB_SNAPSHOT_PATH="chains/testnet/"     # testnet by default
+CHAINSPEC_FILE="testnet_chainspec.json"
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h | --help) # display Help
+            Help
+            exit;;
+        --archivist) # Run as an archivist
+            ARCHIVIST=true
+            shift;;
+        --ip)
+            PUBLIC_IP="$2"
+            shift 2;;
+        --dns)
+            PUBLIC_DNS="$2"
+            shift 2;;
+        -n | --name) # Enter a name
+            NAME="$2"
+            shift 2;;
+        -d | --data_dir) # Choose the data directory
+            HOST_BASE_PATH="$2"
+            shift 2;;
+        --mainnet) # Join the mainnet
+            DB_SNAPSHOT_PATH="chains/mainnet/"
+            CHAINSPEC_FILE="mainnet_chainspec.json"
+            DB_SNAPSHOT_URL="${MAINNET_DB_SNAPSHOT_URL}"
+            CLIAIN_ENDPOINT='wss://ws.azero.dev:443'
+            MAINNET=true
+            shift;;
+        --local) # Bootstrap a local network
```

Comment:
So it is supported actually? Or question really should we support it. I thought this repo is used only for Testnet and Mainnet. If we want to have local chain support, the question is how we'll m anage `local_chainspec.json`.

Reply:
Removed -- this probably should be discussed at some point (it's trickier than it seems so I paused it for now)